### PR TITLE
Add new filters dock panel

### PR DESCRIPTION
**Pull request by cristianonicolai**

### CODE REVIEW

**Comment by cristianonicolai** [comment]
Depends on https://github.com/kiegroup/appformer/pull/191

**Comment by cristianonicolai** [comment]
@nmirasch @bsremac please take a look. Again still quite a lot of code duplicate and limited functionality as I'm not applying any of the filters in the list. Also, just migrating filters as is, for instance, status needs to support multiple values.

**Comment by bsremac** [comment]
@cristianonicolai Overall looks good. Are we planning to have some kind of _Clear All_ option, as we have now?

File: jbpm-wb-common/jbpm-wb-common-client/src/main/java/org/jbpm/workbench/common/client/resources/i18n/Constants.java [hunk 1]
```diff
@@ -98,4 +98,14 @@ String DataSetError(String dataSet,
     String Custom();
 
     String Home();
+
+    String Filters();
+
+    String Select();
+
+    String Refine();
```
Comment (bsremac) [inline jbpm-wb-common/jbpm-wb-common-client/src/main/java/org/jbpm/workbench/common/client/resources/i18n/Constants.java:106]: Is this the word UX team went for in the end? Doesn't sounds too descriptive, doesn't it? :)
Even _Filter by_ looks clearer to me.

Comment (cristianonicolai) [reply jbpm-wb-common/jbpm-wb-common-client/src/main/java/org/jbpm/workbench/common/client/resources/i18n/Constants.java:106]: +1, we can discuss that with Liz, it is in the design and prototype that I'm using as the reference.

**Comment by cristianonicolai** [comment]
@bsremac I believe the clear all option will be incorporated into the table header along with the active filters.

**Comment by bsremac** [comment]
@cristianonicolai Ah, right. I remember that to be the idea.
+1

File: jbpm-wb-common/jbpm-wb-common-client/src/main/java/org/jbpm/workbench/common/client/filters/BasicFiltersViewImpl.html [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+        <form role="form" style="margin-top: 20px;">
+            <h5 class="kie-dock__heading--section" data-field="refine"></h5>
+            <div class="form-group">
+                <select class="selectpicker form-control" data-width="100%" data-field="refine-options"></select>
+            </div>
+            <div class="form-group">
+                <div class="input-group" data-field="filters-input">
+                    <span class="input-group-btn">
+                        <button class="btn btn-link" data-field="filters-input-help" type="button" data-toggle="popover" data-trigger="hover" data-html="true" title="" data-placement="bottom" data-container="body">
+                            <i class="fa fa-info-circle"></i>
```
Comment (nmirasch) [inline jbpm-wb-common/jbpm-wb-common-client/src/main/java/org/jbpm/workbench/common/client/filters/BasicFiltersViewImpl.html:12]: @cristianonicolai There is something different in the css applied, related with the 'info-circle'. In this view appears a border around the button when is clicked...

**Review by nmirasch — APPROVED** [review]
Looks very good. Even the calendar icon at the end of the date filter seems that only closes the popup. I think we can cover that when we use the enter data here. It's a +1 for me

File: jbpm-wb-common/jbpm-wb-common-client/src/main/java/org/jbpm/workbench/common/client/filters/BasicFiltersViewImpl.html (hunk 2 above)
Comment (cristianonicolai) [reply jbpm-wb-common/jbpm-wb-common-client/src/main/java/org/jbpm/workbench/common/client/filters/BasicFiltersViewImpl.html:12]: @nmirasch ok, I will take a look at that.

**Pull request merged**